packages/react-native/src/primitives/PhoneNumberField/types.ts

```diff
@@ -0,0 +1,47 @@
+import { StyleProp, TextStyle } from 'react-native';
+
+import { TextFieldProps, TextFieldStyles } from '../TextField';
+
+export interface PhoneNumberFieldProps
+  extends Omit<TextFieldProps, 'secureTextEntry'> {
+  /**
+   * @description
+   * Sets the default dial code that will be selected on initial render
+   */
+  defaultDialCode?: string;
+
+  /**
+   * @description
+   * Accepts an array of dial codes (strings) used as options in the dial code selector
+   */
+  dialCodes: Array<string>;
```

Comment:
Nit: prefer TS shorthand syntax for arrays:
```suggestion
  dialCodes: string[];
```